### PR TITLE
PMM-7266 Progress bar

### PR DESCRIPTION
**Pull request by tiagomotasantos**
https://github.com/Percona-Lab/pmm-submodules/pull/1444

### CODE REVIEW

File: pmm-app/src/pmm-dbaas/components/DBCluster/DBCluster.test.tsx [hunk 1]
```diff
@@ -24,12 +24,8 @@ describe('DBCluster::', () => {
...
     const root = mount(<DBCluster kubernetes={kubernetesStub} />);
 
-    expect(
-      root
-        .find('[data-qa="cluster-status-failed"]')
-        .at(0)
-        .prop('className'),
-    ).toContain('failed');
+    expect(root.find('[data-qa="cluster-progress-bar"]').length).toBeGreaterThan(0);
+    expect(root.find('[data-qa="cluster-status-error-message"]').length).toBeGreaterThan(0);
```
Comment (fabio-silva) [inline pmm-app/src/pmm-dbaas/components/DBCluster/DBCluster.test.tsx:28]: Can we use `dataQa` here?

File: pmm-app/src/pmm-dbaas/components/DBCluster/DBClusterStatus/DBClusterStatus.tsx [hunk 2]
```diff
@@ -2,31 +2,58 @@ import React, { FC, useMemo } from 'react';
...
-export const DBClusterStatus: FC<DBClusterStatusProps> = ({ status, errorMessage }) => {
+export const DBClusterStatus: FC<DBClusterStatusProps> = ({
+  status,
+  message,
+  finishedSteps,
+  totalSteps,
+}) => {
   const styles = useStyles(getStyles);
   const statusError = status === Status.failed || status === Status.invalid;
+  const showProgressBar = status === Status.changing || statusError;
```
Comment (fabio-silva) [inline pmm-app/src/pmm-dbaas/components/DBCluster/DBClusterStatus/DBClusterStatus.tsx:20]: Just to be sure, the intended here is `(status === Status.changing) || statusError`, right?

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: pmm-app/src/pmm-dbaas/components/DBCluster/DBClusterStatus/DBClusterStatus.tsx (hunk 2 above)
Comment (tiagomotasantos) [reply pmm-app/src/pmm-dbaas/components/DBCluster/DBClusterStatus/DBClusterStatus.tsx:20]: Strict equality has higher precedence than logical OR, no need for grouping. So yes that's the intended!

File: pmm-app/src/pmm-dbaas/components/DBCluster/DBCluster.test.tsx (hunk 1 above)
Comment (tiagomotasantos) [reply pmm-app/src/pmm-dbaas/components/DBCluster/DBCluster.test.tsx]: It was an old file so had to change in the rest of the file too, but done!

Comment (fabio-silva) [reply pmm-app/src/pmm-dbaas/components/DBCluster/DBCluster.test.tsx]: 👍

File: pmm-app/src/pmm-dbaas/components/DBCluster/DBClusterStatus/DBClusterStatus.tsx (hunk 2 above)
Comment (fabio-silva) [reply pmm-app/src/pmm-dbaas/components/DBCluster/DBClusterStatus/DBClusterStatus.tsx:20]: Yep, exactly 😄  👍

**Review by fabio-silva — DISMISSED** [review]

File: pmm-app/src/pmm-dbaas/components/ProgressBar/ProgressBar.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  dataQa,
+}) => {
+  const styles = useStyles(getStyles);
+  const progressBarErrorStyles = useMemo(() => ({
+    [styles.progressBarError]: status === ProgressBarStatus.error,
+  }), [status]);
+  const stepsLabelErrorStyles = useMemo(() => ({
+    [styles.stepsLabelError]: status === ProgressBarStatus.error,
+  }), [status]);
+  const width = totalSteps > 0 ? Math.round((finishedSteps * 100) / totalSteps) : 0;
```
Comment (nicolalamacchia) [inline pmm-app/src/pmm-dbaas/components/ProgressBar/ProgressBar.tsx:21]: nit: No need of grouping here
```suggestion
  const width = totalSteps > 0 ? Math.round(finishedSteps * 100 / totalSteps) : 0;
```

**Review by nicolalamacchia — APPROVED** [review]
LGTM, good job!

File: pmm-app/src/pmm-dbaas/components/ProgressBar/ProgressBar.tsx (hunk 3 above)
Comment (tiagomotasantos) [reply pmm-app/src/pmm-dbaas/components/ProgressBar/ProgressBar.tsx:21]: There is a lint rule about not mixing operators that force me to do this.

**Review by lunaticusgreen — APPROVED** [review]
👍

**Pull request merged**